src/ImageData/CartaFitsImage.cc

```diff
@@ -0,0 +1,1271 @@
+                native_type = casacore::SpectralCoordinate::VRAD;
+            } else if (ctype1.startsWith("BETA")) {
+                native_type = casacore::SpectralCoordinate::BETA;
+            } else {
+                spdlog::debug("Spectral coordinate type {} not supported.", ctype1);
+                wcsfree(&wcs_spectral);
+                return false;
+            }
+
+            // Translate spectral axis to FREQ
+            int spectral_axis_index(0); // in wcs_spectral struct
+            char ctype[9];
+            strcpy(ctype, "FREQ-???");
+            int status = wcssptr(&wcs_spectral, &spectral_axis_index, ctype);
+
+            if (status) {
+                switch (status) {
+                    case 4:
+                    case 5:
+                    case 6:
+                    case 7:
+                        break;
+                    default:
+                        ok = false;
+                }
+            } else {
+                // Set up wcsprm struct with wcsset
+                casacore::Coordinate::set_wcs(wcs_spectral);
+            }
+
+            if (ok) {
+                // Determine frequency type
+                casacore::MFrequency::Types frequency_type = GetFrequencyType(wcs_spectral);
+
+                if (frequency_type == casacore::MFrequency::Undefined) {
+                    spdlog::debug("Failed to determine spectral reference frame.");
+                    wcsfree(&wcs_spectral);
+                    return false;
+                }
+
+                try {
+                    bool one_based(true);
+                    casacore::SpectralCoordinate spectral_coord(frequency_type, wcs_spectral, one_based);
+                    spectral_coord.setNativeType(native_type);
+                    coord_sys.addCoordinate(spectral_coord);
+                } catch (const casacore::AipsError& err) {
+                    // Catch so the wcs sub can be freed
+                    ok = false;
+                }
+            }
+        }
+    }
+
+    wcsfree(&wcs_spectral);
+    return ok;
+}
+
+casacore::MFrequency::Types CartaFitsImage::GetFrequencyType(const ::wcsprm& wcs_spectral) {
+    // Determine frequency type from wcs spectral headers
+    casacore::MFrequency::Types freq_type(casacore::MFrequency::Undefined);
+
+    if (wcs_spectral.specsys[0] == '\0') {
+        // If no SPECSYS, use VELREF
+        if (wcs_spectral.velref == 0) {
+            // No VELREF either, freq type undefined
+            return freq_type;
+        } else {
+            int velref = wcs_spectral.velref;
+            if (velref > 256) {
+                velref -= 256;
+            }
+
+            std::vector<casacore::MFrequency::Types> velref_freq_types = {casacore::MFrequency::LSRK, casacore::MFrequency::BARY,
+                casacore::MFrequency::TOPO, casacore::MFrequency::LSRD, casacore::MFrequency::GEO, casacore::MFrequency::REST,
+                casacore::MFrequency::GALACTO};
```

Comment:
Since this is used only once during image setup, I thought it would be better to create the vector temporarily rather than storing it for the lifetime of the object.